### ISSUE TITLE
Redirect Request - Conditions Team

### ISSUE THREAD

**Issue by tyler-spangler6** [department-of-veterans-affairs/va.gov-team#90086]
### Status
**[2024-08-16] [Fran]** Tyler messaged Randi about pushing this redirect to next week. They don't have our changes into Prod and would like to try again next week so requesting this to be on hold until they update their form URL in the form on vets-website. 
**[2024-08-07] [Fran]** CAIA has approved, and this is good to go. Note that the earliest date to implement is 8/15.
**[2024-08-06] [Fran]** Moved to Next Refinement column to be refined and then moved into Sprint 10 during sprint planning. **_Note that we are awaiting CAIA approval and the nod to release from Tyler._**
**[2024-08-06] [Fran]** Note that the **earliest date to release per below is Aug 15th.**

### Instructions
- Requests must be submitted **AT LEAST** 2 weeks in advance.  Some requests take a significant amount of time and coordination to implement, so start the process as soon as you know you will need one. 
- This issue will be used from initial request through implementation to ensure all individuals working on this are notified of status updates.  Please do not create multiple issues to track different steps.
- It is your responsibility to notify VA stakeholders as appropriate.

| POC | Github alias(es) |
--- | --- 
| Product team point(s) of contact: | @yurenanava , @tyler-spangler6  |
| Content point of contact: | `CAIA will enter` |
| IA point of contact: | @kristinoletmuskat  |

### Type of request
- [X] We are retiring or taking down a page and need to redirect the URL (complete redirect section)
- [ ] We are changing the URL of an existing page (complete redirect section)
- [ ] We need a custom vanity URL (complete vanity URL section)
- [ ] We are removing a temporary redirect  (add relevant background and link to original redirect request)

### Do you have a product technical team able to implement this redirect?
- [X] Yes - Please note: Redirect still needs approval from IA.
  - [ ] Please verify your team has reviewed [Redirect governance](https://design.va.gov/components/url-standards/redirects) and [Redirect technical implementation information
](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/engineering/redirect-implementation-strategy.md) 
- [ ] No - CAIA will need to make Public Websites aware of the ticket.

### Implementation date
When does this request need to be live? **NOTE:** Redirects in vsp-platform-revproxy auto-deploy M-Th. Friday launches should be avoided.
Choose one:
- [ ] On a specific date, connected to a program or publishing requirement. **DATE:**
- [ ] Any day prior to a deadline/latest possible date. **DEADLINE:** 
- [X] Any day after an earliest possible date. **EARLIEST DATE:** Aug 15
- [ ] During a specific date range: ***EARLIEST DATE TO LATEST DATE:***
- [ ] No hard timing requirements, publish when possible.

### Redirects 
Current URL  |  Redirect Destination or New URL
---  |  ---
https://va.gov/disability/file-disability-claim-form-21-526ez/new-disabilities-revised/add | https://va.gov/disability/file-disability-claim-form-21-526ez/new-disabilities/add

### Vanity URLs 
Vanity URL  |  Landing page
---  |  ---
requested URL | landing page URL

**Link to campaign landing page request issue, if relevant:**

### Process, Roles and Responsibilities
- [X] Requesting team: Above information is provided
- [X] Requesting team: All appropriate VA stakeholders are notified as appropriate
- [x] IA: Request is vetted and documented and implementation plan is clear 
- [x] IA: Request is assigned to appropriate team for implementation 
- [ ] Implementation team: Work is complete
- [ ] Implementation team: Validated in production
- [ ] Requesting team: Validates everything is correct in production and closes ticket

**Comment by kristinoletmuskat** [comment]
Approved on the CAIA side! No need to inform any stakeholders for this one.

**Comment by FranECross** [comment]
This hasn't bee refined yet, but CAIA has give approval. Since it seems to be a simple redirect, I pulled it into Ready (although it's not yet pointed), so that it can be pulled into the sprint if needed. cc @randimays @jilladams

**Comment by tyler-spangler6** [comment]
Will these changes be available in production today? Or when will these go out? I am waiting on final approval for our URL update in the form, so just need to know if i need to get these merged to go out today in prod. Thanks for the help! 
cc: @yurenanava

**Comment by tyler-spangler6** [comment]
I messaged Randi about pushing this redirect to next week. We do not have our changes into Prod and would like to try again next week so requesting this to be on hold until we update our form URL in the form on vets-website. Sorry about this!

**Comment by jilladams** [comment]
Can we clarify 2 things?: 

1. when I click the "from" link right now (https://va.gov/disability/file-disability-claim-form-21-526ez/new-disabilities-revised/add): 
 * If unauthed: I get redirected to https://www.va.gov/disability/file-disability-claim-form-21-526ez/introduction
  * If authed: I land on https://www.va.gov/disability/file-disability-claim-form-21-526ez/start

 I'm not sure it's possible for us to redirect from a question page within a form flow to a different questions page -- shouldn't this routing be set up within your React app? 

2. If there's a product team that built the form flow / question pages, could they be the ones to implement this change, whether it happens within React or as a redirect?

**Comment by jilladams** [comment]
(If for some reason it's necessary for Public Websites to make this change as a redirect, also noting: Randi will be on PTO 8/20-8/30, so please be sure to talk to @FranECross for timing considerations, so we can make sure someone else can help, as needed.)

**Comment by jilladams** [comment]
(cc @randimays in case I'm missing something here, since you've already talked to @tyler-spangler6 about this change.)

**Comment by tyler-spangler6** [comment]
@jilladams Thanks for the help with this. The purpose of this redirect is for users who have a saved in progress form and the last URL they visited is `new-disabilities-revised/add` when they log back into their form flow, we want them to be directed to `new-disabilities/add` when we update the URL. If we do not implement this redirect, they are routed to the first page in the form.  If this is something that we should implement in the React app, we can do that, I think I was just confused on the approval process for this. Sorry if i misunderstood this process, but if this sounds like we should implement it in the React form flow, please just let me know and we can start working on adding that.  Thanks again for everything. 
cc: @yurenanava

**Comment by randimays** [comment]
Thanks @jilladams for the callout! I didn't realize this was for a React app. 

Yes, @tyler-spangler6 this should be implemented in React. In order to get more control over the routing in your app (i.e. prevent 404s from deep-diving), you need to add your app endpoint(s) to this file: https://github.com/department-of-veterans-affairs/vsp-platform-revproxy/blob/main/template-rendering/revproxy-vagov/vars/react_routes.yml. Then you can add logic to send the user to a different page if certain conditions are met.

Redirects are a last-resort option as they create a lot of tech debt as time goes on. If we can't change the behavior in the React app for some reason, or if we're sending users from one Drupal-created page to another, or if we're moving across domains, then we implement redirects.

**Comment by tyler-spangler6** [comment]
@randimays Thank you for the help with this and explaining the intent of these. I apologize for my misunderstanding and will start to work to implement this in the vets-website react application. I imagine we can go ahead and close this ticket for now and if we run into needing this we can open a new one.  
@yurenanava

**Comment by randimays** [comment]
No worries, @tyler-spangler6! I'll close this ticket as a no-op.

**Comment by jilladams** [comment]
(Valid question about the approval process for your URL structure. I'm not sure about that, but since CAIA are on the ticket, if y'all have any questions or concerns about Tyler's plan, feel free to reopen for discussion around the IA decisions.)
